autoscheduler/frontend/src/components/NavBar/LoginButton.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import * as Cookies from 'js-cookie';
+import { Button, Typography } from '@material-ui/core';
+import LogoutIcon from '@material-ui/icons/ExitToAppOutlined';
+import reloadPage from './reloadPage';
+import * as styles from './NavBar.css';
+
+// This file contains the login button. When a user logs in, it is replaced by their name
+// followed by the logout button.
+const LoginButton: React.FC = () => {
+  // Used to track whether the API call has returned anything yet
+  const [loading, setLoading] = React.useState(true);
+  // default set to defaultTestUserName instead of empty string becuase I thought
+  // an empty string might cause the refreshes on logout test to report false positives
+  // very easily depending on future changes.
+  const [usersName, setUsersName] = React.useState('defaultTestUserName');
```

Comment:
```suggestion
  const [usersName, setUsersName] = React.useState<string>();
```
Use undefined as a placeholder value instead